public/js/controllers/buildSummary.js

```diff
@@ -0,0 +1,195 @@
+CDash.controller('BuildSummaryController',
+  function BuildSummaryController($scope, $rootScope, $http, renderTimer) {
+
+    // Support for the various graphs on this page.
+    $scope.showTimeGraph = false;
+    $scope.showErrorGraph = false;
+    $scope.showWarningGraph = false;
+    $scope.showTestGraph = false;
+    $scope.showHistoryGraph = false;
+    $scope.graphLoading = false;
+    $scope.graphLoaded = false;
+    $scope.graphData = [];
+
+    $scope.loading = true;
+    $http({
+      url: 'api/v1/buildSummary.php',
+      method: 'GET',
+      params: $rootScope.queryString
+    }).success(function(cdash) {
+      renderTimer.initialRender($scope, cdash);
+      // Set title in root scope so the head controller can see it.
+      $rootScope['title'] = cdash.title;
+    }).finally(function() {
+      $scope.loading = false;
+    });
+
+
+    // Show/hide our various history graphs.
+    $scope.toggleTimeGraph = function() {
+      $scope.showTimeGraph = !$scope.showTimeGraph;
+
+      if (!$scope.graphLoaded) {
+        $scope.loadGraphs();
+      }
+    };
+    $scope.toggleErrorGraph = function() {
+      $scope.showErrorGraph = !$scope.showErrorGraph;
+
+      if (!$scope.graphLoaded) {
+        $scope.loadGraphs();
+      }
+    };
+    $scope.toggleWarningGraph = function() {
+      $scope.showWarningGraph = !$scope.showWarningGraph;
+
+      if (!$scope.graphLoaded) {
+        $scope.loadGraphs();
+      }
+    };
+    $scope.toggleTestGraph = function() {
+      $scope.showTestGraph = !$scope.showTestGraph;
+
+      if (!$scope.graphLoaded) {
+        $scope.loadGraphs();
+      }
+    };
+    $scope.toggleHistoryGraph = function() {
+      $scope.showHistoryGraph = !$scope.showHistoryGraph;
+
+      if (!$scope.graphLoaded) {
+        $scope.loadGraphs();
+      }
+    };
+
+    // Load graph data via AJAX.
+    $scope.loadGraphs = function() {
+      $scope.graphLoading = true;
+      $http({
+        url: 'api/v1/getPreviousBuilds.php',
+        method: 'GET',
+        params: { buildid: $scope.cdash.build.id }
+      }).success(function(resp) {
+        $scope.setupGraphs(resp['builds']);
+        $scope.graphLoaded = true;
+      }).finally(function() {
+        $scope.graphLoading = false;
+      });
+    };
+
+    // Initialize each graph with newly loaded data.
+    $scope.setupGraphs = function (builds) {
+      var buildtime = [];
+      var builderrors = [];
+      var buildwarnings = [];
+      var testfailed = [];
+      var buildids = [];
+
+      $scope.cdash.buildhistory = [];
+
+      // Isolate data for each graph.
+      for (var i = 0, len = builds.length; i < len; i++) {
+        var build = builds[i];
+        var t = new Date(build['starttime']).getTime();
```

Comment:
This returns 'NaN' on my system. 

var t = new Date("2016-09-08 16:09:20").getTime();  <---- NaN

var t = new Date("2016-09-08T16:09:20").getTime();  <--- Ok